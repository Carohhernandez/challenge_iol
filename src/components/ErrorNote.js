import React from 'react';
import { Link } from "react-router-dom";
import styled from 'styled-components';

// SVG
import { ReactComponent as MortyLogo } from '../svg/morty.svg';

const ErrorContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const ErrorTitle = styled.div`
    display: flex;
`;

const ErrorLink = styled(Link)`
    display: block;

    border: 1px solid #fff;
    border-radius: 5px;
    padding: 10px;

    text-decoration: none;
    color: #fff;
`;

const ErrorNote = ({withBack = false, message}) => (
    <ErrorContainer> 
        <ErrorTitle>
            <h1>{message}</h1>
            <MortyLogo />
        </ErrorTitle>
        {withBack && <ErrorLink to="/"> Go back to list </ErrorLink>}
    </ErrorContainer>
);

export default ErrorNote;