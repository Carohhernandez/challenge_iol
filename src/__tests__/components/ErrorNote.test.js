import { render, screen } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import ErrorNote from '../../components/ErrorNote';

describe('Go back link response to prop', () =>{
    test("renders ErrorNote with back button", () => {
        render(
            <MemoryRouter>
                <ErrorNote withBack/>
            </MemoryRouter>
        );
        const backButton = screen.queryByText(/Go back to list/i);
        expect(backButton).toBeInTheDocument();
    });
    
    test("renders ErrorNote without back button", () => {
        render(
            <MemoryRouter>
                <ErrorNote/>
            </MemoryRouter>
        );

        const backButton = screen.queryByText(/Go back to list/i);
        expect(backButton).not.toBeInTheDocument();
    }); 
});

test("renders ErrorNote correctly", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <ErrorNote withBack/>
        </MemoryRouter>
    );
  expect(asFragment()).toMatchSnapshot();
});

