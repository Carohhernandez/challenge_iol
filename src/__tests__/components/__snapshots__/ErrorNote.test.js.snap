// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ErrorNote correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe hAFWbH"
  >
    <div
      class="sc-gswNZR fXKJtf"
    >
      <h1 />
      <svg>
        morty.svg
      </svg>
    </div>
    <a
      class="sc-dkrFOg eBxNmd"
      href="/"
    >
       Go back to list 
    </a>
  </div>
</DocumentFragment>
`;
